### PR TITLE
Multiple servers

### PR DESCRIPTION
**Pull request by mattbsox**
adding ability to specify multiple servers for one liberty runtime in a single build file

### CODE REVIEW

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy [hunk 1]
```diff
@@ -51,181 +33,36 @@ class Liberty implements Plugin<Project> {
...
         //Create expected server extension from liberty extension data
         project.afterEvaluate {
             setEclipseFacets(project)
-            if (project.liberty.server == null) {
+            if (isSingleServerProject(project)) {
+                new LibertySingleServerTasks(project).applyTasks()
+            } else if (isMultiServerProject(project)) {
+                new LibertyMultiServerTasks(project).applyTasks()
+            } else if (project.liberty.server != null && !project.liberty.servers.isEmpty()){
+                throw new GradleException('Both a \'server\' and \'servers\' closure were found in a build.gradle file that uses the liberty plugin. Please define multiple servers inside of the Liberty \'servers\' closure in your build.gradle file.')
```
Comment (dshimo) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy]: refer to server closures as server extensions

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/LibertyMultiServerTasks.groovy [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+        }
+
+        overwriteTask('installLiberty', InstallLibertyTask, {
+            description 'Installs Liberty from a repository'
+            logging.level = LogLevel.INFO
+            group 'Liberty'
+        })
+
+        project.tasks.getByName('libertyRun') {
+            description = "Runs a Websphere Liberty Profile server under the Gradle process."
```
Comment (chyt) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/LibertyMultiServerTasks.groovy]: Change to just say "Liberty"

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/LibertySingleServerTasks.groovy [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            description "Automatically generates arquillian.xml for projects that use Arquillian Liberty Managed or Remote containers."
+            logging.level = LogLevel.INFO
+            group 'Liberty'
+            dependsOn 'libertyCreate', 'processTestResources'
+            skipIfArquillianXmlExists = project.configureArquillian.skipIfArquillianXmlExists
+            arquillianProperties = project.configureArquillian.arquillianProperties
+        })
+
+        if (!dependsOnApps(project.liberty.server)) {
+            if (project.plugins.hasPlugin('war')) {
```
Comment (cherylking) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/LibertySingleServerTasks.groovy]: Need to check for 'ear' plugin also.

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/LibertyMultiServerTasks.groovy [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+        project.tasks.addRule(pattern) { String taskName ->
+            if (taskName.startsWith(name)) {
+                project.task(taskName, type: taskType, overwrite:true) {
+                    server = project.liberty.servers.getByName(taskName - "${name}-")
+                }.configure(configureClosure)
+            }
+        }
+    }
+
+    boolean tasksUpToDate (List<String> taskList) {
```
Comment (dshimo) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/LibertyMultiServerTasks.groovy]: what if we set the upper level's outputs to be the collection of servers' outputs

File: src/main/groovy/net/wasdev/wlp/gradle/plugins/LibertyMultiServerTasks.groovy [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+        taskList.each {
+            if(!project.tasks.getByName(it).getUpToDate()){
+                return false
```
Comment (chyt) [inline src/main/groovy/net/wasdev/wlp/gradle/plugins/LibertyMultiServerTasks.groovy]: Check that this returns from the boolean method and not just its own block

**Pull request merged**